--- links-2.1pre28/intl/polish.lng.orig	2007-03-21 21:55:27.000000000 +0100
+++ links-2.1pre28/intl/polish.lng	2007-04-18 14:45:06.153887469 +0200
@@ -371,11 +371,11 @@
 T_SSL_ERROR, "B³±d SSL",
 T_NO_SSL, "Ta wersja Linksa nie posiada obs³ugi SSL/TLS",
 T_NO_SMB_CLIENT, "Brak zainstalowanego klienta smb (obs³ugiwane s± smbclient i smbc)",
-T_DECOMPRESSION_ERROR, NULL,
-T_ERROR_DECOMPRESSING_, NULL,
-T__wITH_, NULL,
-T_COMPRESSED_ERROR, NULL,
-T_UNKNOWN_COMPRESSION_METHOD, NULL,
+T_DECOMPRESSION_ERROR, "B³±d dekompresji",
+T_ERROR_DECOMPRESSING_, "B³±d dekompresji ",
+T__wITH_, " przy u¿yciu ",
+T_COMPRESSED_ERROR, "B³±d w skompresowanych danych",
+T_UNKNOWN_COMPRESSION_METHOD, "Nieznana metoda kompresji",
 T_SURE_DELETE, "Jeste¶ pewny, ze chcesz usun±æ",
 T_BOOKMARKS_ALREADY_IN_USE, "Zak³adki w u¿yciu!",
 T_ASSOCIATIONS_ALREADY_IN_USE, "Skojarzenia w u¿yciu!",
