--- links-2.8/bookmark.c	2013-09-01 19:13:47.000000000 +0200
+++ links-2.8/bookmark.c.new	2013-10-07 19:23:03.763106002 +0200
@@ -719,6 +719,18 @@
 		safe_strncpy(e, cast_uchar "bookmarks.html", MAX_STR_LEN - (e - bookmarks_file));
 #endif
 	}
+	/* try to create bookmarks.html based on old bookmarks (from links <= 0.97) */
+	if (access(bookmarks_file, R_OK) != 0) {
+		char *prev;
+
+		if ((prev = get_current_dir_name()) && chdir(links_home) == 0) {
+			if (access("bookmarks", R_OK) == 0 && access("/usr/bin/perl", X_OK) == 0) {
+				system("/usr/bin/perl -lne '@l = split(q(\\|)); print qq(<dt><a href=\"$l[1]\">$l[0]</a>)' bookmarks > bookmarks.html");
+			}
+			chdir(prev);
+			free(prev);
+		}
+	}
 
 	bookmark_ld.codepage=utf8_table;
 	load_bookmarks(NULL);
