--- links-2.1pre7/struct.h.orig	Mon Jun 10 01:24:28 2002
+++ links-2.1pre7/struct.h	Thu Dec 26 01:24:58 2002
@@ -82,7 +82,7 @@
 		long mid; /* Jako Mikulasovo ID - to je byro! B;-) */
 		long type;
 		long value;/*tohle se bude casto typecastovat na pointer*/
-		long handler; /* smysl bude mit jen u vnitrnich pnych */
+		long long handler; /* smysl bude mit jen u vnitrnich pnych */
 		long index; /* smysl ma jen u selectitek */
 		struct localnamespace* next;
 	}lns;
--- links-2.1pre7/builtin.c.orig	Sun Sep 22 10:35:38 2002
+++ links-2.1pre7/builtin.c	Thu Dec 26 01:25:16 2002
@@ -757,6 +757,7 @@
 	long*pompointer;
 	int i=0,j=0;
 	time_t*cas;
+	time_t temptime;
 	char pomchar;
 
 	plns*pomns;
@@ -1240,12 +1241,14 @@
 					if(!(pomarg=getarg(&argy)))
 					{/*	pomint=to32int(pomarg,context);*/
 						time_t t = (time_t)pomint1;
+						temptime = t;
 						casek=localtime(&t);
 						/* Je to doba od usvitu dejin a aby toho nebylo malo tak v sekundach :-( */
 					}
 					else
 					{	cas=js_mem_alloc(sizeof(time_t));
 						time(cas);
+						temptime=*cas;
 						casek=localtime(cas);
 						js_mem_free(cas);
 						casek->tm_year=pomint1;
@@ -1264,6 +1267,7 @@
 				}
 			} else{	cas=js_mem_alloc(sizeof(time_t));
 				*cas=time(0);
+				temptime=*cas;
 				casek=localtime(cas);
 				js_mem_free(cas);
 				if(pomarg) delarg(pomarg,context);
@@ -1296,10 +1296,10 @@
 			BUILDFCE("getSeconds",CgetSeconds);
 			pomvar->handler=(long)pompointer;
 			BUILDFCE("setTime",CsetTime);
-			pomvar->handler=casek->tm_sec;
+			pomvar->handler=((long long)temptime) * 1000;
 			pompointer=&pomvar->handler;
 			BUILDFCE("getTime",CgetTime);
-			pomvar->handler=(long)pomns;
+			pomvar->handler=(long)pompointer;
 			BUILDFCE("setYear",CsetYear);
 			pomvar->handler=casek->tm_year;
 			pompointer=&(pomvar->handler);
@@ -2202,8 +2206,9 @@
 		break;
 		case CgetTime:
 			idebug("CgetTime called ");
-			rettype=INTEGER;
-			retval=*(long*)variable->handler;
+			rettype=FLOAT;
+			retval=(long)js_mem_alloc(sizeof(float));
+			*(float*)retval=*(long long*)variable->handler;
 			idebug("and exited!\n");
 		break;
 		case CsetYear:
