diff -urN aaa/links-current/configure.in links-current/configure.in
--- aaa/links-current/configure.in	Thu May 16 08:06:43 2002
+++ links-current/configure.in	Thu May 23 20:57:36 2002
@@ -16,6 +16,7 @@
 #AC_PROG_AWK
 #AM_PROG_LEX
 #AC_PROG_YACC
+AC_PROG_CXX
 
 dnl Check for libraries
 #AC_CHECK_LIB(fl,main,AC_DEFINE(JS) LIBS="$LIBS -lfl",AC_MSG_WARN(You don't have libfl; you won't be able to run javascript))
@@ -83,35 +83,8 @@
 AC_CHECK_SIZEOF(unsigned, "$default_int")
 AC_CHECK_SIZEOF(long, "$default_int")
 AC_CHECK_SIZEOF(unsigned long, "$default_int")
-AC_CACHE_CHECK([for big endian], ac_cv_big_endian,
-	AC_TRY_RUN([
-	long l;
-	char *c = (char *)&l;
-	int main()
-	{
-		l = 0x12345678L;
-		return !(c[[sizeof(long) - 1]] == 0x78 && c[[sizeof(long) - 2]] == 0x56 && c[[sizeof(long) - 3]] == 0x34 && c[[sizeof(long) - 4]] == 0x12);
-	}
-	], ac_cv_big_endian=yes, ac_cv_big_endian=no, ac_cv_big_endian=no)
-)
-AC_CACHE_CHECK([for little endian], ac_cv_little_endian,
-	AC_TRY_RUN([
-	long l;
-	char *c = (char *)&l;
-	int main()
-	{
-		l = 0x12345678L;
-		return !(c[[0]] == 0x78 && c[[1]] == 0x56 && c[[2]] == 0x34 && c[[3]] == 0x12);
-	}
-	], ac_cv_little_endian=yes, ac_cv_little_endian=no, ac_cv_little_endian="$ac_cv_have_emx")
-)
 
-if test "$ac_cv_big_endian" = yes; then
-	AC_DEFINE(C_BIG_ENDIAN)
-else if test "$ac_cv_little_endian" = yes; then
-	AC_DEFINE(C_LITTLE_ENDIAN)
-fi
-fi
+AC_C_BIGENDIAN([AC_DEFINE(C_BIG_ENDIAN)], [AC_DEFINE(C_LITTLE_ENDIAN)])
 
 dnl Checks for library functions.
 AC_PROG_GCC_TRADITIONAL
@@ -401,7 +402,6 @@
 		AC_CHECK_LIB(stdc++, main)
 		AC_CHECK_LIB(atheos, main)
 		if test "$ac_cv_lib_atheos_main" = yes; then
-			AC_PROG_CXX
 			AC_DEFINE(GRDRV_ATHEOS)
 			drivers="$drivers ATHEOS"
 			cf_have_atheos=yes
