--- links-2.1pre16/view.c.orig	2005-01-22 21:51:55.000000000 +0100
+++ links-2.1pre16/view.c	2005-01-28 20:21:11.813229048 +0100
@@ -3112,7 +3112,7 @@
 			goto x;
 		}
 		if ((upcase(ev->x) == 'Q' && !ev->y) || ev->x == KBD_CTRL_C) {
-		  exit_prog(ses->term, (void *)(ev->x == KBD_CTRL_C), ses); /* 2989: warning: cast to pointer from integer of different size */
+		  exit_prog(ses->term, (void *)(ev->x == KBD_CTRL_C || ev->x == 'Q'), ses); /* 2989: warning: cast to pointer from integer of different size */
 			goto x;
 		}
 		if (ev->x == KBD_CLOSE){
