--- links-2.7/bookmarks.c.orig	2013-06-23 18:55:29.428859752 +0200
+++ links-2.7/bookmarks.c	2013-06-23 19:03:09.398840448 +0200
@@ -716,6 +716,18 @@
 	if (!*bookmarks_file)
 		snprintf(cast_char bookmarks_file,MAX_STR_LEN,"%sbookmarks.html",links_home?links_home:(unsigned char*)"");
 
+	/* try to create bookmarks.html based on old bookmarks (from links <= 0.97) */
+	if (access(bookmarks_file, R_OK) != 0) {
+	  char *prev;
+	  if ((prev = get_current_dir_name()) && chdir(links_home) == 0) {
+	    if (access("bookmarks", R_OK) == 0 && access("/usr/bin/perl", X_OK) == 0) {
+	      system("/usr/bin/perl -lne '@l = split(q(\\|)); print qq(<dt><a href=\"$l[1]\">$l[0]</a>)' bookmarks > bookmarks.html");
+	    }
+	    chdir(prev);
+	    free(prev);
+	  }
+	}
+
 	bookmark_ld.codepage=utf8_table;
 	load_bookmarks(NULL);
 }
