--- links-2.1pre16/default.c.orig	2005-01-22 21:51:53.000000000 +0100
+++ links-2.1pre16/default.c	2005-01-28 20:47:14.138719472 +0100
@@ -312,10 +312,10 @@
 			sleep(3);
 			mem_free(home_links);
 			home_links = stracpy(home);
-			add_to_strn(&home_links, ".links");		
+			add_to_strn(&home_links, ".links2");		
 		}
 		mem_free(config_dir);
-	} else add_to_strn(&home_links, ".links");
+	} else add_to_strn(&home_links, ".links2");
 	if (stat(home_links, &st)) {
 #ifdef HAVE_MKDIR
 		if (!mkdir(home_links, 0777)) goto home_creat;
@@ -327,7 +327,7 @@
 	first_failed:
 	mem_free(home_links);
 	home_links = stracpy(home);
-	add_to_strn(&home_links, "links");
+	add_to_strn(&home_links, "links2");
 	if (stat(home_links, &st)) {
 #ifdef HAVE_MKDIR
 		if (!mkdir(home_links, 0777)) goto home_creat;
@@ -390,7 +390,7 @@
 void load_config(void)
 {
 #ifdef SHARED_CONFIG_DIR
-	load_config_file(SHARED_CONFIG_DIR, "links.cfg");
+	load_config_file(SHARED_CONFIG_DIR, "links2/links.cfg");
 #endif
 	load_config_file(links_home, "links.cfg");
 	load_config_file(links_home, "html.cfg");
--- links-2.1pre16/links.1.orig	2005-01-22 21:51:55.000000000 +0100
+++ links-2.1pre16/links.1	2005-01-28 20:47:50.217234704 +0100
@@ -303,7 +303,7 @@
 
 .SH FILES
 .TP
-.IP "~/.links/links.cfg"
+.IP "~/.links2/links.cfg"
 Per-user configfile, automatically created by
 .B links.
 .SH PLATFORMS
