--- links-2.1pre28/file.c.orig	2006-11-28 17:23:14.000000000 +0100
+++ links-2.1pre28/file.c	2007-04-18 14:49:21.420434281 +0200
@@ -235,9 +235,14 @@
 		mem_free(name);
 		setcstate(c, S_FILE_TYPE); abort_connection(c); return;
 	}
+opening:	
 	if ((h = open(name, O_RDONLY | O_NOCTTY)) == -1) {
 		int er = errno;
 		if ((d = opendir(name))) goto dir;
+		if (strncmp(name + strlen(name) - 3, ".gz", 3) != 0) {
+			add_to_strn(&name, ".gz");
+			goto opening;
+		} 
 		mem_free(name);
 		setcstate(c, -er); abort_connection(c); return;
 	}
