diff -urN /tmp/links-0.96/default.c links-0.96/default.c
--- /tmp/links-0.96/default.c	Sun Apr  8 22:46:18 2001
+++ links-0.96/default.c	Wed Mar  6 18:56:09 2002
@@ -910,6 +910,7 @@
 	{1, gen_cmd, str_rd, str_wr, 0, MAX_STR_LEN, bookmarks_file, "bookmarks_file", "bookmarks-file"},
 	{1, gen_cmd, str_rd, str_wr, 0, MAX_STR_LEN, default_anon_pass, "ftp.anonymous_password", "ftp.anonymous-password"},
 	{1, gen_cmd, cp_rd, NULL, 0, 0, &dds.assume_cp, "assume_codepage", "assume-codepage"},
+	{1, gen_cmd, cp_rd, NULL, 0, 0, &dump_codepage, "dump_codepage", "dump-codepage"},
 	{1, NULL, term_rd, term_wr, 0, 0, NULL, "terminal", NULL},
 	{1, NULL, term2_rd, NULL, 0, 0, NULL, "terminal2", NULL},
 	{1, NULL, type_rd, type_wr, 0, 0, NULL, "association", NULL},
diff -urN /tmp/links-0.96/links.h links-0.96/links.h
--- /tmp/links-0.96/links.h	Mon Jun 11 12:14:25 2001
+++ links-0.96/links.h		Wed Mar  6 18:56:09 2002
@@ -1107,6 +1107,8 @@
 
 extern int retval;
 
+extern int dump_codepage;
+
 extern unsigned char *path_to_exe;
 
 void unhandle_terminal_signals(struct terminal *term);
diff -urN /tmp/links-0.96/main.c links-0.96/main.c
--- /tmp/links-0.96/main.c	Sun Apr  8 22:46:06 2001
+++ links-0.96/main.c		Wed Mar  6 18:39:33 2002
@@ -2,6 +2,8 @@
 
 int retval = RET_OK;
 
+int dump_codepage = 0;
+
 void unhandle_basic_signals(struct terminal *);
 
 void sig_terminate(struct terminal *t)
@@ -199,7 +201,7 @@
 		o.xw = 80;
 		o.yw = 25;
 		o.col = 0;
-		o.cp = 0;
+		o.cp = dump_codepage;
 		ds2do(&dds, &o);
 		o.plain = 0;
 		o.frames = 0;
