--- links-2.2/main.c	2007-12-11 17:45:38.000000000 +0100
+++ links-2.2/main.c.new	2010-01-21 22:00:22.020403979 +0100
@@ -4,6 +4,9 @@
  * This file is a part of the Links program, released under GPL.
  */
 
+#define _GNU_SOURCE
+#include <string.h>
+
 #include "links.h"
 
 int retval = RET_OK;
@@ -290,6 +290,7 @@
 
 static void fixup_g(void)
 {
+	if (strncmp(basename(path_to_exe), "glinks", 6) == 0) ggr = 1;
 	if (ggr_drv[0] || ggr_mode[0] || force_g) ggr = 1;
 	if (dmp) ggr = 0;
 }
