--- links-2.1pre16/links.1.orig	2005-01-22 21:51:55.000000000 +0100
+++ links-2.1pre16/links.1	2005-01-28 20:47:50.217234704 +0100
@@ -303,7 +303,7 @@
 
 .SH FILES
 .TP
-.IP "~/.links/links.cfg"
+.IP "~/.links2/links.cfg"
 Per-user configfile, automatically created by
 .B links.
 .SH PLATFORMS

--- links-2.14/default.c.orig	2016-11-16 23:49:10.000000000 +0100
+++ links-2.14/default.c	2017-07-08 17:18:34.278675496 +0200
@@ -749,7 +749,7 @@ skip_path_conv:;
 		while (home_links[0] && dir_sep(home_links[strlen(cast_const_char home_links) - 1])) home_links[strlen(cast_const_char home_links) - 1] = 0;
 		EINTRLOOP(rs, stat(cast_const_char home_links, &st));
 		if (!rs && S_ISDIR(st.st_mode)) {
-			add_to_strn(&home_links, cast_uchar "/links");
+			add_to_strn(&home_links, cast_uchar "/links2");
 		} else {
 			fprintf(stderr, "CONFIG_DIR set to %s. But directory %s doesn't exist.\n\007", config_dir, home_links);
 			portable_sleep(3000);
@@ -761,9 +761,9 @@ skip_path_conv:;
 	} else {
 		add_dot_links:
 #ifdef OPENVMS
-		add_to_strn(&home_links, cast_uchar "links");
+		add_to_strn(&home_links, cast_uchar "links2");
 #else
-		add_to_strn(&home_links, cast_uchar ".links");
+		add_to_strn(&home_links, cast_uchar ".links2");
 #endif
 	}
 	EINTRLOOP(rs, stat(cast_const_char home_links, &st));
@@ -792,7 +792,7 @@ skip_path_conv:;
 #ifdef DOS
 	add_to_strn(&home_links, cast_uchar "links.cfg");
 #else
-	add_to_strn(&home_links, cast_uchar "links");
+	add_to_strn(&home_links, cast_uchar "links2");
 #endif
 	EINTRLOOP(rs, stat(cast_const_char home_links, &st));
 	if (rs) {
