--- links-2.1pre21/intl/polish.lng.orig	2006-03-11 02:44:00.000000000 +0100
+++ links-2.1pre21/intl/polish.lng	2006-04-17 20:58:19.686700750 +0200
@@ -340,7 +340,7 @@
 T_PASSWORD_FOR_ANONYMOUS_LOGIN, "Has³o dla logowania anonimowego",
 T_USE_PASSIVE_FTP, "U¿ywaj pasywnego FTP",
 T_USE_FAST_FTP, "U¿ywaj szybkiego FTP",
-T_SET_TYPE_OF_SERVICE, NULL,
+T_SET_TYPE_OF_SERVICE, "Ustaw typ us³ugi IP (ToS)",
 T_MANUAL, "Podrêcznik u¿ytkownika",
 T_HOMEPAGE, "Strona domowa",
 T_CALIBRATION, "Kalibracja",
