--- links-0.96/cookies.c.pix	Wed Jun  6 18:32:50 2001
+++ links-0.96/cookies.c	Thu Sep  6 14:49:37 2001
@@ -78,7 +79,7 @@
 	struct c_server *cs;
 	unsigned char *p, *q, *s, *server, *document;
 	if (accept_cookies == ACCEPT_NONE) return 0;
-	for (p = str; *p != ';' && *p; p++) if (WHITECHAR(*p)) return 0;
+	for (p = str; *p != ';' && *p && !(WHITECHAR(*p)); p++);
 	for (q = str; *q != '='; q++) if (!*q || q >= p) return 0;
 	if (str == q || q + 1 == p) return 0;
 	if (!(cookie = mem_alloc(sizeof(struct cookie)))) return 0;
