--- links-2.7/html.c.orig	2013-06-23 18:17:51.248954518 +0200
+++ links-2.7/html.c	2013-06-23 18:30:42.988922132 +0200
@@ -1062,7 +1062,54 @@
 				add_to_strn(&al, cast_uchar "]");
 			} else if (usemap) al = stracpy(cast_uchar "[USEMAP]");
 			else if (ismap) al = stracpy(cast_uchar "[ISMAP]");
-			else al = stracpy(cast_uchar "[IMG]");
+			else{
+				unsigned char *str = get_attr_val(a, "src");
+				unsigned char *s;
+				int r, i;
+				/* How images will be displayed:
+				* fake_alt = 0   -- do not truncate long names,
+				* fake_alt = 30  -- truncate long names to 30% of term width,
+				* fake_alt = 100 -- truncate long names to 100% of term width.
+				*/         
+				int fake_alt = 20;
+				int max_len;
+				int name_len;
+				/* substitute string for hidden characters */
+				unsigned char *fake_str = stracpy("*");
+				if(str && fake_alt){
+					/* FIXME: replace following '80' with screen width */
+					max_len = (int)80*((float)fake_alt/100);
+					r = strcspn(str, "?");
+					if (!(s = mem_alloc((r + 1) * sizeof(char)))) return;
+					strncpy(s, str, r);
+					s[r] = '\0';
+					for(r = strlen(s) - 1; r >= 0; --r)
+						if(dir_sep(s[r])) break;
+					r++;
+					if(strlen(s + r) > max_len){
+						for(i = strlen(s) -1; i>=0; --i)
+							if(s[i] == '.') break;
+						if(max_len < strlen(s + i)) al = stracpy("[IMG]");
+						else{
+							if(!(al = mem_alloc((max_len + strlen(fake_str) + 3) * sizeof(char)))) return;
+							name_len = max_len - strlen(s + i);
+							strcpy(al, "[\0");
+							strncat(al, s + r, name_len/2);
+							strcat(al, fake_str);
+							strcat(al, s + r + (strlen(s + r) - max_len + name_len/2));
+							strcat(al, "]");
+						}
+					}
+					else{
+						if(!(al = mem_alloc((strlen(s + r) + 3) * sizeof(char)))) return;
+						sprintf(al, "[%s]", s + r);
+					}
+					mem_free(s);
+					mem_free(str);
+					mem_free(fake_str);
+				}
+				else al = stracpy("[IMG]");
+			}
 		}
 		if (al) {
 			if (ismap) {
