--- links-2.1pre24/intl/polish.lng.orig	2006-10-10 14:22:22.000000000 +0200
+++ links-2.1pre24/intl/polish.lng	2006-11-19 14:05:06.141881750 +0100
@@ -197,7 +197,7 @@
 T_ERROR_DOWNLOADING, "Wyst±pi³ b³±d podczas pobierania",
 T_ERROR_LOADING, "Wyst±pi³ b³±d podczas ³adowania",
 T_COULD_NOT_CREATE_FILE, "Nie mo¿na utworzyæ pliku",
-T_COULD_NOT_CREATE_TEMPORARY_FILE, NULL,
+T_COULD_NOT_CREATE_TEMPORARY_FILE, "Nie mo¿na utworzyæ pliku tymczasowego",
 T_UNKNOWN_TYPE, "Nieznany typ",
 T_CONTEN_TYPE_IS, "Typ zawarto¶ci to",
 T_HAS_TYPE, "ma typ",
