--- links-2.7/default.c.orig	2013-06-23 19:40:46.132079076 +0200
+++ links-2.7/default.c	2013-06-23 19:42:18.822075186 +0200
@@ -1216,14 +1216,14 @@
 {
 	struct driver_param *dp;
 	foreachback(dp, driver_params) {
-		if ((!dp->param || !*dp->param) && !dp->codepage && !*dp->shell) continue;
+		if ((!dp->param || !*dp->param) && !dp->codepage && (!dp->shell || !*dp->shell)) continue;
 		if (dp->nosave) continue;
 		add_nm(o, s, l);
 		add_quoted_to_str(s, l, dp->name);
 		add_to_str(s, l, cast_uchar " ");
 		add_quoted_to_str(s, l, dp->param ? dp->param : (unsigned char*)"");
 		add_to_str(s, l, cast_uchar " ");
-		add_quoted_to_str(s, l, dp->shell);
+		add_quoted_to_str(s, l, dp->shell ? dp->shell : (unsigned char*)"");
 		add_to_str(s, l, cast_uchar " ");
 		add_to_str(s, l, get_cp_mime_name(dp->codepage));
 	}
